src/ResourceManagement/Sql/SqlServerKeyImpl.cs

```diff
@@ -0,0 +1,303 @@
+            : base(innerObject, null)
+        {
+            this.serverKeyName = serverKeyName;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            if (innerObject != null && innerObject.Name != null)
+            {
+                this.serverKeyName = innerObject.Name;
+            }
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="serverKeyName">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses firewall rule operations.</param>
+        ///GENMHASH:137DAACD2261698E82334933A0868D1A:889083AF977CA8FF77804D3045E866B0
+        internal SqlServerKeyImpl(string serverKeyName, ServerKeyInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.serverKeyName = serverKeyName;
+            this.sqlServerManager = sqlServerManager;
+            if (innerObject != null && innerObject.Id != null)
+            {
+                try
+                {
+                    ResourceId resourceId = ResourceId.FromString(innerObject.Id);
+                    this.resourceGroupName = resourceId.ResourceGroupName;
+                    this.sqlServerName = resourceId.Parent.Name;
+                }
+                catch (Exception) { }
+            }
+            if (innerObject != null && innerObject.Name != null)
+            {
+                this.serverKeyName = innerObject.Name;
+            }
+        }
+
+        ///GENMHASH:3E38805ED0E7BA3CAEE31311D032A21C:03B7E74B9CEC907229FC7E3E3FC8EFC4
+        public override string Name()
+        {
+            return this.serverKeyName;
+        }
+
+        ///GENMHASH:69682825EBBC81814AD1914F79080C3A:842E861459B1C0CB42302534E5479B32
+        public SqlServerKeyImpl WithCreationDate(DateTime creationDate)
+        {
+            this.Inner.CreationDate = creationDate;
+            return this;
+        }
+
+        ///GENMHASH:C608B2A3A894C122878626FA09D94406:BE33F46EDC996DB65B7F46EC68568C24
+        public SqlServerKeyImpl WithAzureKeyVaultKey(string uri)
+        {
+            this.Inner.ServerKeyType = Models.ServerKeyType.AzureKeyVault.Value;
+            this.Inner.Uri = uri;
+            // If the key URI is "https://YourVaultName.Vault.Azure.Net/keys/YourKeyName/01234567890123456789012345678901",
+            // then the Server Key Name should be formatted as: "YourVaultName_YourKeyName_01234567890123456789012345678901"
+            string[] items = uri.Split('/');
+            this.serverKeyName = $"{items[2].Split('.')[0]}_{items[4]}_{items[5]}";
```

Comment:
>items[2].Split('.')[0] [](start = 36, length = 22)

did you check this? AFAIK you cannot have  complex calculation in this kind of string formatting. May need to assign this to a variable and then use.

Reply:
yes, I checked ;-)